Shoko.Server/API/v3/Controllers/RenamerController.cs

```diff
@@ -13,39 +14,204 @@ namespace Shoko.Server.API.v3.Controllers
     public class RenamerController : BaseController
     {
+        /// <summary>
+        /// Get a list of all <see cref="Models.Shoko.Renamer"/>s.
+        /// </summary>
+        /// <returns></returns>
         [HttpGet]
-        public ActionResult<List<RenamerInfo>> Index()
+        public ActionResult<List<Models.Shoko.Renamer>> GetAllRenamers()
         {
-            return RenameFileHelper.Renamers.Select(r =>
-            {
-                return new RenamerInfo
-                {
-                    Description = r.Value.description,
-                    Id = r.Key,
-                    Enabled = ServerSettings.Instance.Plugins.EnabledRenamers.ContainsKey(r.Key) ? ServerSettings.Instance.Plugins.EnabledRenamers[r.Key] : true, 
-                    Priority = ServerSettings.Instance.Plugins.Priority.Contains(r.Key) ? ServerSettings.Instance.Plugins.Priority.IndexOf(r.Key) : int.MaxValue,
-                };
-            }).ToList();
+            return RenameFileHelper.Renamers.Values
+                .Select(v => new Models.Shoko.Renamer(v))
+                .ToList();
+        }
+        
+        /// <summary>
+        /// Get the <see cref="Models.Shoko.Renamer"/> by the given <paramref name="renamerID"/>.
+        /// </summary>
+        /// <param name="renamerID">Renamer ID</param>
+        /// <returns></returns>
+        [HttpGet("{renamerID}")]
+        public ActionResult<Models.Shoko.Renamer> GetRenamer([FromRoute] int renamerID)
+        {
+            var renamer = RenameFileHelper.Renamers.Values.FirstOrDefault(r => r.ID == renamerID);
+            if (renamer == null)
+                return NotFound("Renamer not found.");
+
+            return new Models.Shoko.Renamer(renamer);
+        }
+
+        /// <summary>
+        /// Disable the <see cref="Models.Shoko.Renamer"/> by the given <paramref name="renamerID"/>.
+        /// </summary>
+        /// <param name="renamerID">Renamer ID</param>
+        /// <returns></returns>
+        [Authorize("admin")]
+        [HttpPost("{renamerID}/Disable")]
+        public ActionResult DisableRenamerByRenamerID(int renamerID)
+        {
+            var renamer = RenameFileHelper.Renamers.Values.FirstOrDefault(r => r.ID == renamerID);
+            if (renamer == null)
+                return NotFound("Renamer not found.");
+
+            ServerSettings.Instance.Plugins.EnabledRenamers[renamer.Name] = false;
+            ServerSettings.Instance.SaveSettings();
+
+            return Ok();
         }
 
-        [HttpDelete("{renamerID}")]
-        public ActionResult Disable(string renamerID)
+        /// <summary>
+        /// Enable the <see cref="Models.Shoko.Renamer"/> by the given <paramref name="renamerID"/>.
+        /// </summary>
+        /// <param name="renamerID">Renamer ID</param>
+        /// <returns></returns>
+        [Authorize("admin")]
+        [HttpPost("{renamerID}/Enable")]
+        public ActionResult EnableRenamerByRenamerID(int renamerID)
```

Comment:
What does the int ID relate to?
Renamer types are stored by a string key, which is how they are defined in software for consistency reasons.